pandas/io/pytables.py

```diff
@@ -1924,6 +1926,13 @@ def get_atom_coltype(self, kind=None):
         else:
             col_name = "%sCol" % kind.capitalize()
 
+        if kind.startswith('complex'):
```

Comment:
The closure here was implemented to preserve the logic -- this function is expected to return a callable `*Col` type (e.g. `Float64Col`) which can then be used as in `FloatCol(shape=(100,)` or just `FloatCol`.  `ComplexCol works differently since it requires `itemsize=8` or `itemsize=16`, so the returned value can be called with the correct itemsize.